pkg/cmdutil/json_flags.go

```diff
@@ -174,14 +174,18 @@ type jsonExporter struct {
 }
 
 // NewJSONExporter returns an Exporter to emit JSON.
-func NewJSONExporter() Exporter {
+func NewJSONExporter() *jsonExporter {
 	return &jsonExporter{}
 }
 
 func (e *jsonExporter) Fields() []string {
 	return e.fields
 }
 
+func (e *jsonExporter) SetFields(fields []string) {
```

Comment:
Having a getter and setter for the `fields` field felt a bit off but decided this was the quickest way to get this change in. In the future we should evaluate just exporting `jsonExporter` and its fields. Perhaps when doing the refactor I mentioned in the comment below.